agent/stats/container.go

```diff
@@ -96,6 +97,9 @@ func (container *StatsContainer) collect() {
 func (container *StatsContainer) processStatsStream() error {
 	dockerID := container.containerMetadata.DockerID
 	seelog.Debugf("Collecting stats for container %s", dockerID)
+	if container.client == nil {
+		return errors.New("Client is not set.")
```

Comment:
> more context

ok!

> When can this error occur?

During tests when agent doesn't set up go docker clients. Docker clients are set up as part of agent startup. If this line isn't here, it panics instead.

> Was there a code path?

This gets executed as part of a looping goroutine for some of the tests in agent/stats.  However, it only triggers sometimes depending on how fast the tests in agent/stats get cleaned up. Running the race detector makes the tests slower: slow enough to trigger this panic about 50% of the time.